ublox/src/ubx_packets/packets.rs

```diff
@@ -0,0 +1,718 @@
+    reserved3: u32,
+    in_proto_mask: u16,
+    out_proto_mask: u16,
+    flags: u16,
+    reserved5: u16,
+}
+
+/// Port Identifier Number (= 4 for SPI port)
+#[ubx_extend]
+#[ubx(from_unchecked, into_raw, rest_error)]
+#[repr(u8)]
+#[derive(Debug, Copy, Clone)]
+pub enum SpiPortId {
+    Spi = 4,
+}
+
+/// UTC Time Solution
+#[ubx_packet_recv]
+#[ubx(class = 1, id = 0x21, fixed_payload_len = 20)]
+struct NavTimeUTC {
+    /// GPS Millisecond Time of Week
+    itow: u32,
+    time_accuracy_estimate_ns: u32,
+    /// Nanoseconds of second, range -1e9 .. 1e9
+    nanos: i32,
+    /// Year, range 1999..2099
+    year: u16,
+    /// Month, range 1..12
+    month: u8,
+    /// Day of Month, range 1..31
+    day: u8,
+    /// Hour of Day, range 0..23
+    hour: u8,
+    /// Minute of Hour, range 0..59
+    min: u8,
+    /// Seconds of Minute, range 0..59
+    sec: u8,
+    /// Validity Flags
+    #[ubx(map_type = NavTimeUtcFlags)]
+    valid: u8,
+}
+
+#[ubx_extend_bitflags]
+#[ubx(from, rest_reserved)]
+bitflags! {
+    /// Validity Flags of `NavTimeUTC`
+    pub struct NavTimeUtcFlags: u8 {
+        /// Valid Time of Week
+        const VALID_TOW = 1;
+        /// Valid Week Number
+        const VALID_WKN = 2;
+        /// Valid UTC (Leap Seconds already known)
+        const VALID_UTC = 4;
+    }
+}
+
+/// Navigation/Measurement Rate Settings
+#[ubx_packet_send]
+#[ubx(class = 6, id = 8, fixed_payload_len = 6)]
+struct CfgRate {
+    /// Measurement Rate, GPS measurements are taken every `measure_rate_ms` milliseconds
+    measure_rate_ms: u16,
+    /// Navigation Rate, in number of measurement cycles.
+    /// On u-blox 5 and u-blox 6, this parametercannot be changed, and is always equals 1.
+    nav_rate: u16,
+    /// Alignment to reference time
+    #[ubx(map_type = AlignmentToReferenceTime)]
+    time_ref: u16,
+}
+
+/// Alignment to reference time
+#[repr(u16)]
+#[derive(Clone, Copy, Debug)]
+pub enum AlignmentToReferenceTime {
+    Utc = 0,
+    Gps = 1,
+}
+
+impl AlignmentToReferenceTime {
+    const fn into_raw(self) -> u16 {
+        self as u16
+    }
+}
+
+/// Set Message Rate the current port
+#[ubx_packet_send]
+#[ubx(class = 6, id = 1, fixed_payload_len = 3)]
+struct CfgMsg3 {
+    msg_class: u8,
+    msg_id: u8,
+    /// Send rate on current Target
+    ///
+    rate: u8,
+}
+
+impl CfgMsg3Builder {
+    #[inline]
+    pub fn set_rate_for<T: UbxPacketMeta>(rate: u8) -> Self {
+        Self {
+            msg_class: T::CLASS,
+            msg_id: T::ID,
+            rate,
+        }
+    }
+}
+
+/// Set Message rate configuration
+/// Send rate is relative to the event a message is registered on.
+/// For example, ifthe rate of a navigation message is set to 2,
+/// the message is sent every secondnavigation solution
+#[ubx_packet_send]
+#[ubx(class = 6, id = 1, fixed_payload_len = 8)]
+struct CfgMsg8 {
+    msg_class: u8,
+    msg_id: u8,
+    /// Send rate on I/O Port (6 Ports)
+    rates: [u8; 6],
+}
+
+impl CfgMsg8Builder {
```

Comment:
> CfgMsgRateBuilder

This name derived from packet name: CfgMsg8, so you suggest call "packet" with `CfgMsgRate` name?
But this would be against naming scheme - use the same names for packets as in ubx protocol documentation.

Reply:
I guess really the question is, where are the names `CfgMsg8` and `CfgMsg3` coming from? Should they be called `CfgMsgSinglePort` and `CfgMsgAllPorts`?

Reply:
> where are the names CfgMsg8 and CfgMsg3

From payload length, it should be unique. So I used it as suffix. But Single/AllPort looks better, and I renamed them.